modules/nextflow/src/main/groovy/nextflow/scm/RepositoryProvider.groovy

```diff
@@ -87,6 +90,14 @@ abstract class RepositoryProvider {
      */
     abstract String getRepositoryUrl()
 
+    /**
+     * @return a org.eclipse.jgit.transport.CredentialsProvider object for authenticating git operations
+     * like clone, fetch, pull, and update
+     **/
+    protected CredentialsProvider getGitCredentials() {
```

Comment:
We can make this public (which is the default permission in groovy) therefore `protected` can be omitted.

Reply:
done